extract/extract/utils.py

```diff
@@ -107,6 +117,14 @@ def sustainability_labels_to_certificates(
                 result.update({certificate_string: CertificateType.UNKNOWN})  # type: ignore[attr-defined] # noqa
                 logger.info(f"unknown sustainability label: {certificate_string}")
 
+    # check extracted certificates for category-specific version
+    if product_category in _certificate_category_names.keys():
+        for certificate_string, certificate in result.items():
+            for label in SUSTAINABILITY_LABELS.keys():
+                for category_alt_name in _certificate_category_names.get(product_category, []):
+                    if re.search(f"^{certificate.value}.*{category_alt_name}$", label):
+                        result.update({certificate_string: label})
```

Comment:
Regarding the mypy issue, it seems that it's complaining about the fact that `certificate` can be None,
so maybe after `for certificate_string, certificate in result.items():` add 
```
if certificate: 
    for label in SUSTAINABILITY_LABELS.keys() ....
```

Reply:
I think this is a false positive from mypy, so I went with ignoring this error. 
In Line 113-117 we actually make sure that `None` is assigned to `Certificate.Unknown`: https://github.com/calgo-lab/green-db/blob/8feaf241c3b590369de1b444291660d4dfbadcb0/extract/extract/utils.py#L113-L117